bindings/java/src/main/com/apple/foundationdb/EventKeeper.java

```diff
@@ -0,0 +1,178 @@
+     * 
+     * @param event the event to get the time for
+     * @return  the total time measured for this event, in nanoseconds. If the event was never recorded,
+     * return 0 instead.
+     */
+    long getTimeNanos(Event event);
+
+    /**
+     * Get the amount of time taken by this event, in the specified units.
+     * 
+     * Important note: If the time that was measured in nanoseconds does not evenly divide the unit that
+     * is specified (which is likely, considering time), then some precision may be lost in the conversion. Use
+     * this carefully.
+     * 
+     * @param event the event to get the time for
+     * @param theUnit the unit to get time in
+     * @return  the total time measured for this event, in the specified unit. If the event was never recorded,
+     * return 0.
+     */
+    default long getTime(Event event, TimeUnit theUnit){
+        return theUnit.convert(getTimeNanos(event),TimeUnit.NANOSECONDS);
+    }
+
+    /**
+     * Marker interface for tracking the specific type of event that occurs, and metadata about said event.
+     * 
+     * Implementations should be sure to provide a quality {@code equals} and {@code hashCode}.
+     */
+    interface Event {
+        /**
+         * @return the name of this event, as a unique string. This name should generally be unique, because 
+         * it's likely that {@code EventKeeper} implementations will rely on this for uniqueness.
+         */
+        String name();
+
+        /**
+         * @return true if this event represents a timed event, rather than a counter event. 
+         */
+        default boolean isTimeEvent() {
+            return false;
+        };
+    }
+
+    /**
+     * An enumeration of static events which occur within the FDB Java driver.
+     */
+    enum Events implements Event {
+        /**
+         * The number of JNI calls that were exercised.
+         */
+        JNI_CALL,
+
+        /**
+         * The total number of bytes fetched over the network, from
+         * {@link Transaction#get(byte[])}, {@link Transaction#getKey(KeySelector)},
+         * {@link Transaction#getRange(KeySelector, KeySelector)} (and related method
+         * overrides), or any other read-type operation which occurs on a Transaction
+         */
+        BYTES_FETCHED,
+
+        /**
+         * The number of times a DirectBuffer was used to transfer a range query chunk
+         * across the JNI boundary
+         */
+        RANGE_QUERY_DIRECT_BUFFER_HIT,
+        /**
+         * The number of times a range query chunk was unable to use a DirectBuffer to
+         * transfer data across the JNI boundary
+         */
+        RANGE_QUERY_DIRECT_BUFFER_MISS,
+        /**
+         * The number of direct fetches made during a range query
+         */
+        RANGE_QUERY_FETCHES,
+        /**
+         * The number of tuples fetched during a range query
+         */
+        RANGE_QUERY_TUPLES_FETCHED,
```

Comment:
Tuples in FoundationDB usually refer to usages of the tuple layer, while this is tracking the more general number of keys read. Can it be changed to something like `RANGE_QUERY_KEYS_FETCHED`?

Reply:
I think `KEYS_FETCHED` would imply that this counter includes counts from e.g. `getKey()` operations (which maybe excessively careful). I switched it to `RANGE_QUERY_RECORDS_FETCHED` instead, which hopefully disambiguates it.